client/core/core.go

```diff
@@ -1765,6 +1779,48 @@ func (c *Core) CreateWallet(appPW, walletPW []byte, form *WalletForm) error {
 	return nil
 }
 
+// createSeededWallet initializes a seeded wallet with an asset-specific seed
+// derived deterministically from the app seed and a random password. The
+// password is returned for encrypting and storing.
+func (c *Core) createSeededWallet(assetID uint32, crypter encrypt.Crypter, form *WalletForm) ([]byte, error) {
+	creds := c.creds()
+	if creds == nil {
+		return nil, fmt.Errorf("no v2 credentials stored")
+	}
+
+	root, err := crypter.Decrypt(creds.EncSeed)
+	if err != nil {
+		return nil, fmt.Errorf("seed decryption error: %w", err)
+	}
+
+	c.log.Infof("Initializing a built-in %s wallet", unbip(assetID))
+
+	b := make([]byte, len(root)+4)
+	copy(b, root)
+	assetB := make([]byte, 4)
+	binary.BigEndian.PutUint32(assetB, assetID)
+	copy(b[len(root):], assetB)
```

Comment:
This is updated in the other PR, but this can just be:

```go
b := make([]byte, len(root)+4)
copy(b, root)
binary.BigEndian.PutUint32(b[len(root):], assetID
```

https://github.com/decred/dcrdex/pull/1230#discussion_r728421901